Added DataAccess Class and renamed Unit Test Class

pjlplourde opened the pull request:
Added SQLite DataAccess class to library along with NuGet packages for Dapper and SQLite; added Unit Test Class and scaffold of test methods.

RudolfJan commented:
Hi Pierre,

Thanks for your contribution. I invited you to be a contributor to the project. I must say I was surprised to see your pull request.  I think it is helpful if you create an issue to describe what you intend to do and assign the issue to yourself. This helps to avoid multiple people working on the same issue. 

I noticed that I cannot view you code change proposals in Visual Studio. This is probably caused because you did not use branch to develop the code. I suggest you change it to work with a branch.  For me this project is also about learning to use Git and Github better in collaboration with other people.

You may want to read this article:

https://gist.github.com/Chaser324/ce0505fbed06b947d962

I tend to disagree with your test cases. You should not throw an exception from a test, that does not make sense. Instead, I think your implemented methods should throw an exception. In the test you check if they throw this exception. Tim Corey shows how to do this in his video on unit testing.

Also I think we should not do merges to the master from code that is not functional. I think in TDD (which I also need to learn), a not implemented method is a step in the development process. You may want to commit that in the branch, but not promote it to the master.


Please, realize that I am learning things as well, so if you disagree with my point of vview, let's discuss it. That should increase the fun.

Regards,

Rudolf

pjlplourde commented:
Hi Rudolf,

Thanks for the comments regarding issues; I'm new to using Git to collaborate, so I'm sorry if I stepped on toes.

If you look at the actual commit (https://github.com/RudolfJan/SQLiteManagerApp/pull/2/commits/ed810ddcea72c37890d14a5237223b9216ee8a53) you will see that the reason you can't see a diff is because the DataAccess.cs file is brand new (you did not have a class file in your class library project), but the commit as well as the files changed (https://github.com/RudolfJan/SQLiteManagerApp/pull/2/files) of the pull request shows the contents of the new file.  Has nothing to do with not having a separate branch.

Generally, there is no need to have a separate branch when working on the first version. The main purpose of a separate branch is to keep your production code isolated from your development code, allowing you to develop in a CI/CD environment without affecting your production environment. The other purpose is when there are multiple developers working on different new features to be added, all from the same base code.

Regarding the unit tests, the first comment is the way Git diffs the files when you change the file name is annoying, because it records it as having deleted the "old" file and added a "new" file.  In this case, your file was called UnitTests1, but just like we generally delete Class1 in a class library, we delete (or rename) UnitTests1 in a Unit Test project, generally to match the class name that is being tested.  In this case, your UnitTests1 class only had an empty Test1 method, so I renamed the file (and the class) to match the new class in the class library: DataAccessTests.  Similarly, methods in the test class should match the name of the method in the class to be tested, so instead of Test1, the DataAccessTests class now has 4 methods, one corresponding to each of the 4 methods in the DataAccess class.

As for the throw new NotImplementedException() line, you'll see in my places in Tim's videos where he talks about this.  While you are absolutely correct that an **implemented** test method should not throw an error, I did implement the methods, hence the throw new NotImplementedException(), indicating that the method body has yet to be written.  This is why my commit message was "**scaffold** of the test methods" and not "**wrote** the test methods".

Does that make better sense?

Thanks,
Pierre
